app/liquid/liquid_i18n.rb

```diff
@@ -0,0 +1,12 @@
+class I18n::TranslationMissing < Exception; end
+
+module LiquidI18nRails
+  def t(string)
+    return I18n.t(string) unless Rails.env.development? || Rails.env.test?
+    begin
+      I18n.t(string, :raise => true)
+    rescue I18n::MissingTranslationData => e
+      raise I18n::TranslationMissing.new(e.message)
```

Comment:
The reason I did this is because the liquid renderer catches `StandardError`, and `I18n::MissingTranslationData` is a subclass of `StandardError` (as it should be). `TranslationMissing` is subclassed on `Exception`, which makes it higher priority. This way instead of their error bubbling up through the call stack and getting caught above, I catch it here and swap it for one that will actually make it to the top.

That is some hackery that should 100% be documented in a comment and I will do it. Thanks for the catch!